sdcm/provision/azure/kms_provider.py

```diff
@@ -0,0 +1,84 @@
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2025 ScyllaDB
+import logging
+from dataclasses import dataclass
+
+from sdcm.utils.azure_utils import AzureService
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class KmsProvider:
```

Comment:
This is highly specific to Azure, I would name it as such to distinguish between providers.

Reply:
Right, that makes sense, will change it appropriately. :+1:

Reply:
+1